homeassistant/components/tesla/sensor.py

```diff
@@ -61,40 +61,32 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit_of_measurement of the device."""
-        return self._unit
+        return self.units
 
     async def async_update(self):
         """Update the state from the sensor."""
         _LOGGER.debug("Updating sensor: %s", self._name)
         await super().async_update()
         units = self.tesla_device.measurement
 
-        if self.tesla_device.bin_type == 0x4:
+        if self.tesla_device.type == "temperature sensor":
             if self.type == "outside":
                 self.current_value = self.tesla_device.get_outside_temp()
             else:
                 self.current_value = self.tesla_device.get_inside_temp()
             if units == "F":
-                self._unit = TEMP_FAHRENHEIT
+                self.units = TEMP_FAHRENHEIT
             else:
-                self._unit = TEMP_CELSIUS
-        elif self.tesla_device.bin_type == 0xA or self.tesla_device.bin_type == 0xB:
+                self.units = TEMP_CELSIUS
+        elif self.tesla_device.type in ["range sensor", "mileage sensor"]:
             self.current_value = self.tesla_device.get_value()
-            tesla_dist_unit = self.tesla_device.measurement
-            if tesla_dist_unit == "LENGTH_MILES":
-                self._unit = LENGTH_MILES
+            tesla_distunits = self.tesla_device.measurement
```

Comment:
I meant that on line 70 we save `self.tesla_device.measurement` into `units` variable. So we don't need to save it again into `tesla_distunits` here.

Reply:
Oh sorry, makes sense. Converted. Since we're cleaning up the code. I also made a change to use the internal util.distance.convert instead of doing it manually. Let me know if you'd prefer I revert and submit a new PR.